### DISCUSSION TITLE
How to add placeholder value in upstream path to header in downstream path?

### DISCUSSION THREAD

**Discussion by S-sona** [ThreeMammals/Ocelot#1863] [Q&A] [answered]
I need to take a part of the route like `api/service/{headerval1}/{headerval2}` and create a header value that gets passed to the downstream services. Values in `headerval1` and `headerval2` should get passed as headers in downstream service

I have given below route 
```json
{
      "DownstreamPathTemplate": "/shopping/payment/verify/",
      "DownstreamScheme": "http",
      "UpstreamPathTemplate": "/shopping/payment/verify/{headerval1}/{headerval2}",
      "UpstreamHttpMethod": [ "POST", "Options" ],
      "ServiceName": "shopping",
      "SwaggerKey": "shopping",
      "DelegatingHandlers": [
        "PathToHeaderDelegatingHandler"
      ]
},
```
Can someone help me with the delegating handler?

**Comment by S-sona** [comment]
can anyone help me with this issue ?

**Reply by raman-m** [reply]
Oh, my dear!... 
I will help you! :wink:

**Comment by raman-m** [comment, marked as answer]
Dear Author,

You need to use a combination of 3 features
- [Headers Transformation](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html)
- [Placeholders](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html#placeholders)
- [Delegating Handlers](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html)

Currently Ocelot doesn't transform from placeholder to a header except limited number of [system placeholders](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html#placeholders).
We can discuss this functionality, and consider it as a feature... **Will you contribute with a PR?**

But we believe that writing custom delegating handler to convert URL path parts into a headers should be pretty easy for you.
